src/utils/flags.cpp

```diff
@@ -37,6 +40,19 @@ class flag_data
         }                                                                                          \
         break
 
+#define FLAG_DATA_UPDATE_CASE(type, type_enum, suffix)                                             \
+    case type_enum:                                                                                \
+        type tmpval_##type_enum;                                                                   \
+        if (!dsn::buf2##suffix(val, tmpval_##type_enum)) {                                         \
+            return error_s::make(ERR_INVALID_PARAMETERS, fmt::format("{} in invalid", val));       \
+        }                                                                                          \
+        value<type>() = tmpval_##type_enum;                                                        \
+        break
+
+#define FLAG_DATA_UPDATE_STRING()                                                                  \
+    case FV_STRING:                                                                                \
+        return error_s::make(ERR_NO_PERMISSION, "string modifications are not supported")
```

Comment:
NO_PERMISSION literally means you have no authorization to operate something. Use ERR_INVALID_PARAMETERS instead, which is existing code.